src/promptflow/promptflow/_sdk/_service/utils/utils.py

```diff
@@ -36,29 +44,62 @@ def wrapper(*args, **kwargs):
 
 
 def get_port_from_config(create_if_not_exists=False):
-    (HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE).touch(mode=read_write_by_user(), exist_ok=True)
-    with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "r", encoding=DEFAULT_ENCODING) as f:
-        service_config = load_yaml(f) or {}
-        port = service_config.get("service", {}).get("port", None)
-    if not port and create_if_not_exists:
-        with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "w", encoding=DEFAULT_ENCODING) as f:
-            # Set random port to ~/.promptflow/pf.yaml
-            port = get_random_port()
-            service_config["service"] = service_config.get("service", {})
-            service_config["service"]["port"] = port
-            dump_yaml(service_config, f)
-    return port
+    global pfs_port_lock
+    pfs_port_lock.acquire()
+    try:
+        (HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE).touch(mode=read_write_by_user(), exist_ok=True)
+        with open(HOME_PROMPT_FLOW_DIR / PF_SERVICE_PORT_FILE, "r", encoding=DEFAULT_ENCODING) as f:
+            service_config = load_yaml(f) or {}
```

Comment:
what will be `service_config` looks like? maybe share an example in the pr desc? or this comment?

Reply:
offline sync with clement, each env will have it's own PF_SERVICE_PORT_FILE in `.promptflow/pfs/{envname_hashcode}`, service_config of each env is the same as before, will added in the description later